File: Web_francisco/Sprint 2_(Advanced Web)/sistema-tareas-ti/backend/src/main/java/com/taskmanager/model/Task.java
// src/main/java/com/taskmanager/model/Task.java
package com.taskmanager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String title;
    
    private String description;
    
    private boolean critical;
    
    @Enumerated(EnumType.STRING)
    private TaskStatus status = TaskStatus.PENDING;
    
    private LocalDateTime createdAt = LocalDateTime.now();
    
    private LocalDateTime completedAt;
    
    @ManyToOne
    @JoinColumn(name = "assigned_to")
    @JsonIgnore
    private User assignedTo;
}

File: Web_francisco/Sprint 2_(Advanced Web)/sistema-tareas-ti/frontend/src/models/Task.ts
// src/models/Task.ts
export enum TaskStatus {
    PENDING = 'PENDING',
    ASSIGNED = 'ASSIGNED',
    COMPLETED = 'COMPLETED'
  }
  
  export interface Task {
    id: number;
    title: string;
    description: string;
    critical: boolean;
    status: TaskStatus;
    createdAt: string;
    completedAt?: string;
    assignedToId?: number;
    assignedToName?: string;
  }
  